File: CarePackage/Software/BorderlessGaming.cs
namespace CarePackage.Software;

public class BorderlessGaming : BaseSoftware
{
    public override string           Key            { get; set; } = "borderless-gaming";
    public override string           Name           { get; set; } = "Borderless Gaming";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Utility;
    public override string           DownloadName   { get; set; } = "BorderlessGaming.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = true;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.borderless_gaming;
    public override string           Homepage       { get; set; } = "https://github.com/codeusa/borderless-gaming";
    
    private readonly GitHubService _github;

    public BorderlessGaming(IServiceProvider services)
    {
        _github = services.GetRequiredService<GitHubService>();
    }
    
    public override async Task<string> GetDownloadUrlAsync(CancellationToken ct)
    {
        var assets = await _github.GetLatestRepositoryReleaseAssetsAsync("Codeusa", "Borderless-Gaming", ct);
        var asset  = assets.FirstOrDefault(a => a.EndsWith(".exe"));
        
        DownloadUrlResolveException.ThrowIf(asset is null);

        return asset;
    }
}

File: src/lib/software/borderless_gaming.py
from PySide6.QtCore import Slot
from src.lib.software import BaseSoftware, SoftwareCategory
from src.lib.github_release_scraper import GithubReleaseScraper

class BorderlessGaming(BaseSoftware):
    def __init__(self):
        super().__init__()

        self._gh = GithubReleaseScraper('Codeusa', 'Borderless-Gaming', self)
        self._gh.releases_scraped.connect(self._on_releases_scraped)

        self.key = 'borderless-gaming'
        self.name = 'Borderless Gaming'
        self.category = SoftwareCategory.Utility
        self.download_name = 'BorderlessGaming.exe'
        self.is_archive = False
        self.should_cache_url = True
        self.requires_admin = False
        self.icon = 'borderless-gaming.png'
        self.homepage = 'https://github.com/codeusa/borderless-gaming'

    @Slot(list)
    def _on_releases_scraped(self, releases: list[str]):
        asset = next((release for release in releases if release.endswith('.exe')), None)
        if asset:
            self.url_resolved.emit(asset)
        else:
            self.url_resolve_error.emit(self.ResolveError.GitHubAssetNotFoundError)

    def resolve_download_url(self):
        self._gh.get_repo_releases()
